propagators/awsxray/awsxray_propagator.go

```diff
@@ -0,0 +1,181 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	traceHeaderKey       = "X-Amzn-Trace-Id"
+	traceHeaderDelimiter = ";"
+	kvDelimiter          = "="
+	traceIDKey           = "Root"
+	sampleFlagKey        = "Sampled"
+	parentIDKey          = "Parent"
+	traceIDVersion       = "1"
+	traceIDDelimiter     = "-"
+	isSampled            = "1"
+	notSampled           = "0"
+
+	traceFlagNone           = 0x0
+	traceFlagSampled        = 0x1 << 0
+	traceIDLength           = 35
+	traceIDDelimitterIndex1 = 1
+	traceIDDelimitterIndex2 = 10
+	traceIDFirstPartLength  = 8
+	sampledFlagLength       = 1
+)
+
+var (
+	empty                  = trace.SpanContext{}
+	errInvalidTraceHeader  = errors.New("invalid X-Amzn-Trace-Id header value, should contain 3 different part separated by ;")
+	errMalformedTraceID    = errors.New("cannot decode trace id from header, should be a string of hex, lowercase trace id can't be all zero")
+	errInvalidSpanIDLength = errors.New("invalid span id length, must be 16")
+)
+
+// AWS X-Ray propagator serializes Span Context to/from AWS X-Ray headers
+//
+// AWS X-Ray format
+//
+// X-Amzn-Trace-Id: Root={traceId};Parent={parentId};Sampled={samplingFlag}
+type Xray struct{}
```

Comment:
Calling this `Propagator` would provide the user with a more descriptive set of statements instead of the stuttered call. E.g

```
_ := new(aws.Propagator)
```

Reply:
Good idea, thanks!